dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show json;
+import 'dart:html' as html;
+
+import 'package:macrobenchmarks/src/web/bench_text_out_of_picture_bounds.dart';
+
+import 'src/web/bench_draw_rect.dart';
+import 'src/web/bench_simple_lazy_text_scroll.dart';
+import 'src/web/bench_text_out_of_picture_bounds.dart';
+import 'src/web/recorder.dart';
+
+typedef RecorderFactory = Recorder Function();
+
+final Map<String, RecorderFactory> benchmarks = <String, RecorderFactory>{
+  BenchDrawRect.benchmarkName: () => BenchDrawRect(),
+  BenchTextOutOfPictureBounds.benchmarkName: () => BenchTextOutOfPictureBounds(),
```

Comment:
question: the developers are going to add the benchmarks here as they add new ones?

suggestion: shall we query a directory/file, get the names from there?

Reply:
Trouble is that this file runs in the browser, so it doesn't have access to the file system and it needs to import the benchmarks in order to map from the name to the benchmark constructor. The files in `dev/devicelab` are the exact opposite: they _do_ have access to the file system but they cannot import benchmark code. Any ideas?

Reply:
Done.